zrml/prediction-markets/src/tests.rs

```diff
@@ -3122,131 +3130,405 @@ fn it_resolves_a_disputed_market() {
     });
 }
 
-#[test_case(MarketStatus::Active; "active")]
-#[test_case(MarketStatus::CollectingSubsidy; "collecting_subsidy")]
-#[test_case(MarketStatus::InsufficientSubsidy; "insufficient_subsidy")]
-#[test_case(MarketStatus::Closed; "closed")]
-#[test_case(MarketStatus::Proposed; "proposed")]
-#[test_case(MarketStatus::Resolved; "resolved")]
-fn dispute_fails_unless_reported_or_disputed_market(status: MarketStatus) {
-    ExtBuilder::default().build().execute_with(|| {
-        // Creates a permissionless market.
-        simple_create_categorical_market(
-            Asset::Ztg,
+#[test]
+fn it_resolves_a_disputed_court_market() {
```

Comment:
Don't get me wrong. I see your point. 

The test was requested by @maltekliemann for a complete integration test on court outside of the pallet. But I see your point too. I think we shouldn't remove it now, since it was already written, has pretty useful checks, tests court from a different perspective and shows the full process of court from the prediction markets perspective.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/976#pullrequestreview-1356494891

> I think you should write at least one functional test (in prediction-markets) which shows that the entire system functions well. This isn't good style and shouldn't be necessary, but since the behavior of the DisputeApi and what prediction-markets expects of it is so underspecified, I'd feel better if we had a functional test like that.

Reply:
I agree that those kind of tests are useful and I agree we should not remove them. Just pointing out that this specific test is an integration test which would belong to it's own crate instead of living inside the unit tests of the PM pallet